x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/esql/use_data_visualizer_esql_data.tsx

```diff
@@ -0,0 +1,627 @@
+import { filterFields } from '../../../common/components/fields_stats_grid/filter_fields';
+import { IndexBasedDataVisualizerExpandedRow } from '../../../common/components/expanded_row/index_based_expanded_row';
+import { useESQLFieldStatsData } from './use_esql_field_stats_data';
+import type { NonAggregatableField } from '../../types/overall_stats';
+import { ESQLQuery, isESQLQuery } from '../../search_strategy/requests/esql_utils';
+import { DEFAULT_BAR_TARGET } from '../../../common/constants';
+import { type Column, useESQLOverallStatsData } from './use_esql_overall_stats_data';
+import { type AggregatableField } from '../../types/esql_data_visualizer';
+import { useDataVisualizerKibana } from '../../../kibana_context';
+import { DATA_VISUALIZER_GRID_EMBEDDABLE_TYPE } from '../../embeddables/grid_embeddable/constants';
+import type {
+  ESQLDataVisualizerGridEmbeddableInput,
+  ESQLDataVisualizerIndexBasedAppState,
+} from '../../embeddables/grid_embeddable/types';
+import { getDefaultPageState } from '../../constants/index_data_visualizer_viewer';
+
+const defaultSearchQuery = {
+  match_all: {},
+};
+
+const FALLBACK_ESQL_QUERY: AggregateQuery = { esql: '' };
+const DEFAULT_SAMPLING_OPTION: SamplingOption = {
+  mode: 'random_sampling',
+  seed: '',
+  probability: 0,
+};
+
+const defaults = getDefaultPageState();
+
+export const getDefaultESQLDataVisualizerListState = (
+  overrides?: Partial<ESQLDataVisualizerIndexBasedAppState>
+): Required<ESQLDataVisualizerIndexBasedAppState> => ({
+  pageIndex: 0,
+  pageSize: 25,
+  sortField: 'fieldName',
+  sortDirection: 'asc',
+  visibleFieldTypes: [],
+  visibleFieldNames: [],
+  limitSize: '10000',
+  searchString: '',
+  searchQuery: defaultSearchQuery,
+  searchQueryLanguage: SEARCH_QUERY_LANGUAGE.KUERY,
+  filters: [],
+  showDistributions: true,
+  showAllFields: false,
+  showEmptyFields: false,
+  probability: null,
+  rndSamplerPref: 'off',
+  ...overrides,
+});
+export const useESQLDataVisualizerData = (
+  input: ESQLDataVisualizerGridEmbeddableInput,
+  dataVisualizerListState: ESQLDataVisualizerIndexBasedAppState,
+  setQuery?: React.Dispatch<React.SetStateAction<ESQLQuery>>
+) => {
+  const [lastRefresh, setLastRefresh] = useState(0);
+  const { services } = useDataVisualizerKibana();
+  const { uiSettings, fieldFormats, executionContext } = services;
+
+  const parentExecutionContext = useObservable(executionContext?.context$);
+
+  const embeddableExecutionContext: KibanaExecutionContext = useMemo(() => {
+    const child: KibanaExecutionContext = {
+      type: 'visualization',
+      name: DATA_VISUALIZER_GRID_EMBEDDABLE_TYPE,
+      id: input.id,
+    };
+
+    return {
+      ...(parentExecutionContext ? parentExecutionContext : {}),
+      child,
+    };
+  }, [parentExecutionContext, input.id]);
+
+  useExecutionContext(executionContext, embeddableExecutionContext);
+
+  const _timeBuckets = useTimeBuckets();
```

Comment:
I see this is using a duplicate of the timebuckets from the old `time_buckets.js`. That has been replaced by `time_buckets_service.ts` - all places using the old code will be updated to use the new service.

Suggest the use of `timeBucketsService` in https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/public/application/util/time_buckets_service.ts#L17 to reduce duplication and ensure we're using the same service everywher

Reply:
Just realized this is in the data visualizer plugin so you can't reuse and this will need a bigger, separate effort - ignore that comment for now. Sorry for the noise!